.travis.yml

```diff
@@ -10,30 +10,35 @@ cache:
 matrix:
   include:
   - node_js: "10"
-    env: TEST_ORACLEDB=true DB="mssql mysql mysql2 postgres sqlite3 oracledb" CXX=g++-4.8 KNEX_TEST_TIMEOUT=60000 ORACLE_HOME=/u01/app/oracle/product/11.2.0/xe ORACLE_SID=XE OCI_LIB_DIR=/u01/app/oracle/product/11.2.0/xe/lib LD_LIBRARY_PATH=/u01/app/oracle/product/11.2.0/xe/lib
+    env: DB="mssql mysql mysql2 postgres sqlite3" KNEX_TEST_TIMEOUT=60000
   - node_js: "8"
-    env: TEST_ORACLEDB=true DB="mssql mysql mysql2 postgres sqlite3 oracledb" CXX=g++-4.8 KNEX_TEST_TIMEOUT=60000 ORACLE_HOME=/u01/app/oracle/product/11.2.0/xe ORACLE_SID=XE OCI_LIB_DIR=/u01/app/oracle/product/11.2.0/xe/lib LD_LIBRARY_PATH=/u01/app/oracle/product/11.2.0/xe/lib
-  - node_js: "6"
-    env: TEST_ORACLEDB=true DB="mssql mysql mysql2 postgres sqlite3 oracledb" CXX=g++-4.8 KNEX_TEST_TIMEOUT=60000 ORACLE_HOME=/u01/app/oracle/product/11.2.0/xe ORACLE_SID=XE OCI_LIB_DIR=/u01/app/oracle/product/11.2.0/xe/lib LD_LIBRARY_PATH=/u01/app/oracle/product/11.2.0/xe/lib
+    env: DB="mssql mysql mysql2 postgres sqlite3" KNEX_TEST_TIMEOUT=60000
   - node_js: "7"
-    env: DB="mssql mysql mysql2 postgres sqlite3" CXX=g++-4.8 KNEX_TEST_TIMEOUT=60000
-
-install: npm i
+    env: DB="mssql mysql mysql2 postgres sqlite3" KNEX_TEST_TIMEOUT=60000
+  - node_js: "6"
+    env: DB="mssql mysql mysql2 postgres sqlite3" KNEX_TEST_TIMEOUT=60000
+  - node_js: "10"
+    env: TEST_ORACLEDB=true NODE_VER=10 KNEX_TEST_TIMEOUT=60000
```

Comment:
Why split conf for same env into two lines?

Reply:
To not have to run first all normal tests and then start building oracledb docker image, run source building and run all tests one more time just for oracle. Though I have been thinking of committing prebuilt base image for running oracledb tests to dockerhub so that setting up that container would be lot faster.

Reply:
aa I think now I got what you meant... true.

Reply:
I'll try it out a bit different way.

Reply:
Either of these didn't work:

```
  - node_js: "10"
    env: 
    - DB="mssql mysql mysql2 postgres sqlite3" KNEX_TEST_TIMEOUT=60000
    - TEST_ORACLEDB=true NODE_VER=10 KNEX_TEST_TIMEOUT=60000
    - TEST_ORACLEDB=true NODE_VER=8 KNEX_TEST_TIMEOUT=60000
    - TEST_ORACLEDB=true NODE_VER=6 KNEX_TEST_TIMEOUT=60000
```

```
matrix:
  include:
  - node_js: "10"
    env: DB="mssql mysql mysql2 postgres sqlite3" KNEX_TEST_TIMEOUT=60000
    env: TEST_ORACLEDB=true NODE_VER=10 KNEX_TEST_TIMEOUT=60000
    env: TEST_ORACLEDB=true NODE_VER=8 KNEX_TEST_TIMEOUT=60000
    env: TEST_ORACLEDB=true NODE_VER=6 KNEX_TEST_TIMEOUT=60000
  - node_js: "8"
    env: DB="mssql mysql mysql2 postgres sqlite3" KNEX_TEST_TIMEOUT=60000
  - node_js: "7"
    env: DB="mssql mysql mysql2 postgres sqlite3" KNEX_TEST_TIMEOUT=60000
  - node_js: "6"
    env: DB="mssql mysql mysql2 postgres sqlite3" KNEX_TEST_TIMEOUT=60000
```

Reply:
that matrix notation works a bit different way than normal travis config, but it is necessary to be able to setup different environment variables for different node versions.

Reply:
@elhigu
What about
```
  - node_js: "10"
    env: DB="mssql mysql mysql2 postgres sqlite3" KNEX_TEST_TIMEOUT=60000
    env: TEST_ORACLEDB=true NODE_VER=10 KNEX_TEST_TIMEOUT=60000
  - node_js: "8"
    env: DB="mssql mysql mysql2 postgres sqlite3" KNEX_TEST_TIMEOUT=60000
    env: TEST_ORACLEDB=true NODE_VER=8 KNEX_TEST_TIMEOUT=60000
  - node_js: "7"
    env: DB="mssql mysql mysql2 postgres sqlite3" KNEX_TEST_TIMEOUT=60000
  - node_js: "6"
    env: DB="mssql mysql mysql2 postgres sqlite3" KNEX_TEST_TIMEOUT=60000
    env: TEST_ORACLEDB=true NODE_VER=6 KNEX_TEST_TIMEOUT=60000
```
?

Reply:
It won't work either, because only the last env under each section is going to be accounted for (last key overrides earlier one with the same name in yaml)

Reply:
Ah, I see :)

Reply:
Also I think it is easier to read when all oracle envs are one after each other compared to sprinkling them under many sections.